### PR TITLE
Download on setup

### PR DESCRIPTION
**Pull request by bstrauch24**
When setting up the API for the first time, the package github.com/HackIllinois/api does not exist. This change installs it with `go get` when `make setup` is run.

### CODE REVIEW

**Comment by PradyumnaShome** [comment]
How will you run this without first cloning the repository?
`go get -t github.com/HackIllinois/api` does it, right?

**Comment by bstrauch24** [comment]
This is my first time setting up the API since last summer and a lot has changed. I cloned the API and tried to run `make setup` but that failed since the `github.com/HackIllinois/api` wasn't in my `GOPATH`... am I supposed to clone the project into my `GOPATH`?

**Comment by ASankaran** [comment]
This is likely due to the experimental introduction of gomodules in go 1.11.

If you are running a version of go <1.11, then you must use `$GOPATH`. So the makefile will complain that `github.com/HackIllinois/api` is missing if you do not obtain the repo with `go get github.com/HackIllinois/api/...`.

If you are running go 1.11 or go 1.12, then go modules has been experimentally introduced. By default, if you clone the repo into a folder under `$GOPATH` the go compiler will attempt to use the pre-1.11 way of obtaining packages. If you cloned the repo outside of your `$GOPATH` then, the go compiler will use the new modules way of obtaining packages.

See https://blog.golang.org/modules2019 for an overview of the introduction of gomodules.

The API is designed to be used with modules, but in theory will work with pre-1.11. (Automated builds are done with `GO111MODULE=on`, to force the usage of go modules regardless of the repo's location in the filesystem.)


tl;dr

Install go version >=1.11 and clone the repo outside your `$GOPATH`. Then makefile should "just work".

**Comment by bstrauch24** [comment]
Thanks, works in version 1.12!
